modules/diskq/qdisk.c

```diff
@@ -1037,8 +1037,6 @@ qdisk_save_state(QDisk *self, GQueue *qout, GQueue *qbacklog, GQueue *qoverflow)
         return FALSE;
     }
 
-  memcpy(self->hdr->magic, self->file_id, 4);
```

Comment:
I think we should keep this here for now. Changing disk-buffer type is somewhat possible.
However, I think we could keep the magic setting in `_create_header()` too.

Reply:
Nice catch! Fixed.